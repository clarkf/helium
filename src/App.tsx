import Connect from "./app/Connect";

import { useApp } from "./state";

export default function App() {
  const isConnected = useApp((s) => s.connection != null);
  const isSyncing = useApp((s) => Object.keys(s.rooms).length < 1);
  const state = useApp((s) => ({ rooms: s.rooms, connection: s.connection }));

  if (!isConnected) {
    return <ConnectScreen />;
  }

  if (isSyncing) {
    return <SyncingScreen />;
  }

  // We're connected and have synced
  return (
    <FullScreen>
      <h4>State</h4>
      <code>
        <pre>{JSON.stringify(state, null, 4)}</pre>
      </code>
    </FullScreen>
  );
}

function ConnectScreen(): JSX.Element {
  return (
    <FullScreen>
      <h1>Welcome to Helium</h1>
      <Connect />
    </FullScreen>
  );
}

function SyncingScreen(): JSX.Element {
  return <p>Please hold, initial synchronization is in process.</p>;
}

function FullScreen({ children }: React.PropsWithChildren): JSX.Element {
  return (
    <div className="full-screen">
      <div>{children}</div>
    </div>
  );
}
