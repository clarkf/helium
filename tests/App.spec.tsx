// @vitest-environment happy-dom

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";

import App from "../src/App";

describe("<App />", () => {
  it("renders", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
