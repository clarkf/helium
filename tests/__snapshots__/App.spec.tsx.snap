// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> > renders 1`] = `
<DocumentFragment>
  <div
    class="full-screen"
  >
    <div>
      <h1>
        Welcome to Helium
      </h1>
      <form>
        <div>
          <label
            for="connect-host"
          >
            Host
          </label>
          <input
            id="connect-host"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="connect-user"
          >
            User ID
          </label>
          <input
            id="connect-user"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="connect-password"
          >
            Password
          </label>
          <input
            id="connect-password"
            type="password"
            value=""
          />
        </div>
        <div>
          <button
            type="submit"
          >
            Connect
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
